python/IAM_PERMISSION_BOUNDARY_CHECK/IAM_PERMISSION_BOUNDARY_CHECK.py

```diff
@@ -0,0 +1,484 @@
+Then: Return NON_COMPLAINT
+
+Scenario 4:
+Given: IAM users present in Account
+And: Permission boundary policies present in Account
+And: IAM user does have permission boundary attached
+Then: Return COMPLAINT
+
+Scenario 5:
+Given: Valid Rule paramter PermissionBoundaryName provided
+And: IAM users present in Account
+And: IAM user does not have permission boundary attached
+Then: Return NON_COMPLAINT
+
+Scenario 6:
+Given: Valid Rule paramter PermissionBoundaryName provided
+And: IAM users present in Account
+And: IAM user does have permission boundary attached 
+And: The Permission Boundary attached to user is not the one listed in parameter.
+Then: Return NON_COMPLAINT
+
+Scenario 7:
+Given: Valid Rule paramter PermissionBoundaryName provided
+And: IAM users present in Account
+And: IAM user does have permission boundary attached
+And: The Permission Boundary attached to user is the one listed in parameter.
+Then: Return COMPLAINT
+
+Scenario 12:
+Given: Rule paramter PermissionBoundaryName provided
+And: Not Valid one
+Then: Return ERROR
+
+"""
+
+import json
+import datetime
+import boto3
+import botocore
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    """Form the evaluation(s) to be return to Config Rules
+
+    Return either:
+    None -- when no result needs to be displayed
+    a string -- either COMPLIANT, NON_COMPLIANT or NOT_APPLICABLE
+    a dictionary -- the evaluation dictionary, usually built by build_evaluation_from_config_item()
+    a list of dictionary -- a list of evaluation dictionary , usually built by build_evaluation()
+
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    configuration_item -- the configurationItem dictionary in the invokingEvent
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+
+    ###############################
+    # Add your custom logic here. #
+    ###############################
+
+    compliance_results = []
+    if(configuration_item):

```

Comment:
Makes the rule too complex to do Configuration Change and periodic. Limit to periodic. In this case `configuration_item` is always empty